pkg/utils/time.go

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Open Component Model contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/open-component-model/ocm/pkg/errors"
+	"github.com/open-component-model/ocm/pkg/generics"
+)
+
+func ParseDeltaTime(s string, past bool) (time.Time, error) {
```

Comment:
perhaps add a short documentation, as the parameters do not make it obvious how this function is supposed to be used